File: app/src/main/java/com/zwq65/unity/data/network/retrofit/response/enity/Video.java
package com.zwq65.unity.data.network.retrofit.response.enity;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by zwq65 on 2017/08/29
 */

public class Video  implements Parcelable {
    /**
     * _id : 59906c33421aa90f4919c7e1
     * createdAt : 2017-08-13T23:11:47.518Z
     * desc : DC最快的闪电侠，在你看标题时就已跑到银河尽头
     * publishedAt : 2017-08-14T17:04:50.266Z
     * source : chrome
     * type : 休息视频
     * url : http://www.bilibili.com/video/av13207512/
     * used : true
     * who : LHF
     */

    private String _id;
    private String createdAt;
    private String desc;
    private String publishedAt;
    private String source;
    private String type;
    private String url;
    private boolean used;
    private String who;

    protected Video(Parcel in) {
        _id = in.readString();
        createdAt = in.readString();
        desc = in.readString();
        publishedAt = in.readString();
        source = in.readString();
        type = in.readString();
        url = in.readString();
        used = in.readByte() != 0;
        who = in.readString();
    }

    public static final Creator<Video> CREATOR = new Creator<Video>() {
        @Override
        public Video createFromParcel(Parcel in) {
            return new Video(in);
        }

        @Override
        public Video[] newArray(int size) {
            return new Video[size];
        }
    };

    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public String getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(String publishedAt) {
        this.publishedAt = publishedAt;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public boolean isUsed() {
        return used;
    }

    public void setUsed(boolean used) {
        this.used = used;
    }

    public String getWho() {
        return who;
    }

    public void setWho(String who) {
        this.who = who;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(_id);
        dest.writeString(createdAt);
        dest.writeString(desc);
        dest.writeString(publishedAt);
        dest.writeString(source);
        dest.writeString(type);
        dest.writeString(url);
        dest.writeByte((byte) (used ? 1 : 0));
        dest.writeString(who);
    }
}

File: app/src/main/java/com/zwq65/unity/data/network/retrofit/response/enity/Video.kt
/*
 *    Copyright [2017] [NIRVANA PRIVATE LIMITED]
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package com.zwq65.unity.data.network.retrofit.response.enity

import android.annotation.SuppressLint
import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

/**
 * ================================================
 * <p>
 * Created by NIRVANA on 2017/08/30
 * Contact with <zwq651406441@gmail.com>
 * ================================================
 */
@SuppressLint("ParcelCreator")
@Parcelize
open class Video(var _id: String?, var createdAt: String?, var desc: String?, var publishedAt: String?,
                 var source: String?, var type: String?, var url: String?,
                 var isUsed: Boolean = false, var who: String?, var image: Image?) : Parcelable {
    /**
     * _id : 59906c33421aa90f4919c7e1
     * createdAt : 2017-08-13T23:11:47.518Z
     * desc : DC最快的闪电侠，在你看标题时就已跑到银河尽头
     * publishedAt : 2017-08-14T17:04:50.266Z
     * source : chrome
     * type : 休息视频
     * url : http://www.bilibili.com/video/av13207512/
     * used : true
     * who : LHF
     */
}
